src/libponyrt/mem/heap.c

```diff
@@ -71,17 +87,90 @@ static void clear_chunk(chunk_t* chunk, uint32_t mark)
   chunk->shallow = mark;
 }
 
+static void final_small(chunk_t* chunk, uint32_t mark)
+{
+  // run any finalisers that need to be run
+  void* p = NULL;
+
+  uint32_t finalisers = chunk->finalisers;
+  uint32_t bit = 0;
+
+  // if there's a finaliser to run for a used slot
+  while((finalisers != 0) && (0 != (bit = __pony_ctzl(finalisers)))) {
+    p = chunk->m + (bit << HEAP_MINBITS);
+
+    // run finaliser
+    assert((*(pony_type_t**)p)->final != NULL);
```

Comment:
@Praetonus thanks for catching these. I should have caught that but lost track of it due to copy/paste from the sendence ponyc fork.

I'll fix these and the CI failures.